File: cs50/week_2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

//Functions to use
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int count_grade(string text);

//Main
int main(void)
{
    //Get a text from user
    string text = get_string("Text: ");

    //Get variables for Coleman-Liau index
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    int index = count_grade(text);

    //Print
    if (index <= 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 2 && index <= 15)
    {
        printf("Grade %i\n", index);
    }
    else
    {
        printf("Grade 16+\n");
    }
}

//To count letters
int count_letters(string text)
{
    int letters = 0;

    for (int l = 0, len = strlen(text); l < len; l++)
    {
        if (isalpha(text[l]))
        {
            letters += 1;
        }
    }
    return letters;
}

//To count words
int count_words(string text)
{
    int words = 1;

    for (int w = 0, len = strlen(text); w < len; w++)
    {
        if (text[w] == ' ')
        {
            words += 1;
        }
    }
    return words;
}

//To count sentences
int count_sentences(string text)
{
    int sentences = 0;

    for (int s = 0, len = strlen(text); s < len; s++)
    {
        if (text[s] == '.')
        {
            sentences += 1;
        }
        else if (text[s] == '!')
        {
            sentences += 1;
        }
        else if (text[s] == '?')
        {
            sentences += 1;
        }
    }
    return sentences;
}

//Index coleman-liau
int count_grade(string text)
{
    //get variables
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    //run index
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;
    float I = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(I);
    return index;
}

File: cs50/week_6/sentimental-readability/readability.py
from cs50 import get_string


def main():

    # Get text from user
    text = get_string("Text: ")

    # Calculate index
    grade = count_grade(text)

    # Print grade
    if grade <= 1:
        print("Before Grade 1")
    elif grade >= 2 and grade <= 15:
        print(f"Grade {grade}")
    else:
        print("Grade 16+")


def count_letters(text):
    # Declare variable letters
    letters = 0

    # Count letters
    for i in range(len(text)):
        if ord(text[i]) >= 65 and ord(text[i]) <= 90:
            letters += 1
        elif ord(text[i]) >= 97 and ord(text[i]) <= 122:
            letters += 1

    return letters


def count_words(text):
    # Declare variable words
    words = 1

    # Count words
    for i in range(len(text)):
        if ord(text[i]) == 32:
            words += 1

    return words


def count_sentences(text):
    # Declare variable sentences
    sentences = 0

    # Count sentences
    for i in range(len(text)):
        if ord(text[i]) == 33 or ord(text[i]) == 46 or ord(text[i]) == 63:
            sentences += 1

    return sentences


def count_grade(text):
    # Get variables for Coleman-Liau
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    L = letters / words * 100
    S = sentences / words * 100

    # Calculate grade
    I = 0.0588 * L - 0.296 * S - 15.8
    I = round(I)
    return I


main()